File: hw/cañarte/u2/HW18ZooInPython/ec_edu_espe_zoo_view/Zoo.py
from datetime import datetime
import sys
import os
# Saray Cañarte, codecreators, DCCO - ESPE

sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from ec_edu_espe_zoo_model.Cage import Cage
from ec_edu_espe_zoo_model.Food import Food
from ec_edu_espe_zoo_model.Lion import Lion
from ec_edu_espe_zoo_model.Monkey import Monkey
from ec_edu_espe_zoo_model.Shark import Shark

if __name__ == "__main__":
    print("Saray's Zoo \n")
    
    cage_felines = Cage(1, "FelinesCage")
    cage_monkeys = Cage(2, "MonkeysCage")
    cage_birds = Cage(3, "BirdsCage")
    pool = Cage(4, "Pool for Shark")
    
    food = Food(1, "Fish")
    
    animals = []
    
    lion_pedro = Lion(True, 1, "Lion Pedro", datetime(2024, 10, 10), 'f', 0, cage_felines)
    lion_pedro.register(cage_felines)
    animals.append(lion_pedro)
    lion_pedro.feed(2)
    lion_pedro.feed_with_food(3, food)
    print("Lion -->", lion_pedro)
    
    monkey_bear = Monkey(2, "Bear", datetime(2024, 5, 3), 'f', 0, cage_monkeys)
    monkey_bear.feed(5)
    animals.append(monkey_bear)
    print("Animal (monkey) -->", monkey_bear)
    
    lion_juancho = Lion(False, 2, "Lion Juancho", datetime(2024, 5, 3), 'm', 0, cage_felines)
    lion_juancho.feed(4)
    animals.append(lion_juancho)
    print("Animal (lion) -->", lion_juancho)
    
    shark_pepe = Shark(1, "Shark Pepe", datetime(2024, 6, 7), 'm', 0, pool)
    shark_pepe.feed(6)
    animals.append(shark_pepe)
    print("Animal (shark) -->", shark_pepe)
    
    print("\nAnimals in Saray's Zoo -->")
    for animal in animals:
        print(animal)

File: hw/segarra/u2/HW18ZooInPythonAndJavaScript/ec.edu.zooinjavascript.view/Zoo.js

import Cage from "../ec.edu.zooinjavascript.model/Cage.js";
import Food from "../ec.edu.zooinjavascript.model/Food.js";
import Condor from "../ec.edu.zooinjavascript.model/Condor.js";
import Eagle from "../ec.edu.zooinjavascript.model/Eagle.js";
import Shark from "../ec.edu.zooinjavascript.model/Shark.js";
import Monkey from "../ec.edu.zooinjavascript.model/Monkey.js";
import Lion from "../ec.edu.zooinjavascript.model/Lion.js";
import Tiger from "../ec.edu.zooinjavascript.model/Tiger.js";
import Animal from "../ec.edu.zooinjavascript.model/Animal.js";

console.log("Eduardo's Zoo!");

const cageFelines = new Cage(1, "felinesCage");
const cageMonkeys = new Cage(2, "monkeysCage");
const cageBirds = new Cage(3, "birdsCage");
const pool = new Cage(4, "shark tank");
const food = new Food(1, "fish");

const lion = new Lion(1, 'Lion Juancho', new Date(2004, 7, 11), 'm', cageFelines, 223, true);
lion.feed(1, food);
lion.register();

let animal = new Animal;

animal = new Condor(2, "Condor andino", new Date(2004, 7, 11), "f", cageBirds, 156, true);
animal.feed();
animal.register();
console.log(`this animal is --> ${animal.getDescriptionProtected()}`);

animal = new Monkey(3, "Macaco", new Date(2014, 9, 24) , "m", cageMonkeys, 183, true);
animal.feed();
animal.register();
console.log(`this animal is --> ${animal.getDescriptionProtected()}`);

animal = new Shark(4, "Baby shark", new Date(2005, 8, 16), "m", pool, 130, true);
animal.feed();
animal.register();
console.log(`this animal is --> ${animal.getDescriptionProtected()}`);

animal = new Tiger(5, "Tigre toño", new Date(2016, 2, 2), "m", cageFelines, 230, true);
animal.feed();
animal.register();
console.log(`this animal is --> ${animal.getDescriptionProtected()}`);

animal = new Eagle(6, "Aguily", new Date(2022, 6, 7), "f", cageBirds, 134, true);
animal.feed();
animal.register();
console.log(`this animal is --> ${animal.getDescriptionProtected()}`);